### ISSUE TITLE
Bounds check fail for global raster

### ISSUE THREAD

**Issue by BielStela** [nmandery/h3ronpy#48]
Hi! sometimes I run into the same issue with global raster that trigger the `Input array spans more than the bounds of WGS84 - input needs to be in WGS84 projection with lat/lon coordinates` exception. In this example I'm using a [global poopulation raster from sedac](https://sedac.ciesin.columbia.edu/data/set/gpw-v4-population-density-rev11/data-download) that has this shape and transform:

```
>>> src.shape
(4320, 8640)
>>> src.transform
Affine(0.0416666666666667, 0.0, -180.0,
       0.0, -0.0416666666666667, 89.99999999999994)
```
Looking at the source `check_wgs84_bounds` I can't see any issue with the checks but when I do the computation manually the floating point curse strikes:

```
>>> transform.a * shape[1]
360.0000000000003
>>> transform.a * shape[0]
180.00000000000014
```
Both values will fail the check in `check_wgs84_bounds`. I did not debug the rust code but I bet something like this is happening under the hood. Do you know any workaround to do on the the user inputs to avoid this issue (like casting the transform to ints or something)? What I normally do is clipping a few border pixels but it is not ideal. 

Thank you!!

**Comment by nmandery** [comment]
Seems to me the `check_wgs84_bounds` function is a bit to naive. I guess additionally using pythons [`math.isclose`](https://docs.python.org/3/library/math.html#math.isclose) in this check would resolve the floating-point curse there.

In the rust code the raster should be tiled correctly - at least I hope so ;) The `check_wgs84_bounds` function was primarily intended as a quick sanity check to save users a bit debugging time when rasters in other projections are accidentally fed into the library.

**Comment by BielStela** [comment]
Awesome, thanks for the fix @nmandery !

**Comment by nmandery** [comment]
I just released v0.20.2 with the fix.
